### PR TITLE
Update `coffeescript` 

### PR DESCRIPTION
**Pull request by confused-Techie**
This PR unlike my previous attempt at this kind of task over on #100 tries not to get to ambitious with changes.

Following best practices I've changed any usage of `coffee-script` to `coffeescript`. Simply changing what we were installing and any requires, the version has stayed the same and should introduce zero changes.

~~Additionally I've updated `color` to it's most recent version. By doing this we are able to simply (ever so slightly) `src/color.js` by removing a function that would output the Hex value of the color. This is because on newer versions of `color` this is a built in feature, so we could just return what is provided by that application, rather than figure it out ourselves.~~

~~Very likely there is more that can be done to reduce complexity in this file by relying on the module more. But this is a fine start.~~

Since the changes needed to bumping and utilizing `color` are not as straightforward as I thought, the commits have been reverted in this PR. And instead this PR focuses only on fixing usage of `coffeescript`.

In the future I would like to take another look at better utilizing `color` as a library though. And in the meantime this PR if/when merged should likely be squashed to cleanup the messy commit history.

### CODE REVIEW

File: src/color.js [hunk 1]
```diff
@@ -83,9 +83,8 @@ module.exports = class Color {
 
   // Essential: Returns a {String} in the form `'#abcdef'`.
   toHexString() {
-    return `#${numberToHexString(this.red)}${numberToHexString(
-      this.green
-    )}${numberToHexString(this.blue)}`;
+    let colorObj = ParsedColor.rgb(this.red, this.green, this.blue);
+    return colorObj.hex();
```
Comment (confused-Techie) [inline src/color.js:87]:
```suggestion
    return (colorObj.hex()).toUpperCase();
```

Looks like originally Atom would convert these as uppercase parts of text, but the `colors` package does not. This should be enough to resolve the issue

**Comment by confused-Techie** [comment]
So it looks like I'm still having some test failures here.

Namely around the `Color.toHex` function.

While I did swap that out in this PR since it seemed straightforward, with the minor fix I attempted to apply the tests still aren't passing.

What I'll likely go ahead and do is remove taking advantage of the new functions of Color and replace this with a comment.

Since we still do want the upgrades, but the replacement in functionality will likely need to be fleshed out a bit more

**Comment by confused-Techie** [comment]
So it turns out the needs of properly bumping `color` are more significant than I thought, while being still acheiveable. But instead I'll just leave that out of this PR, and let this PR focus only on fixing usage `coffeescript` within the codebase.

**Review by DeeDeeG — APPROVED** [review]
Diff looks good to me!

(The merge conflict is in `yarn.lock`, which makes sense with all the recent dependency change PRs landing. A note on these conflicts: If people will commit updated `yarn.lock` at the same time with their dependency bumps in `package.json`, the chance of `yarn.lock` conflicts goes _down_ a lot, but admittedly not to zero. I'd totally recommend it, though.)

**Comment by confused-Techie** [comment]
> Diff looks good to me!
> 
> (The merge conflict is in `yarn.lock`, which makes sense with all the recent dependency change PRs landing. A note on these conflicts: If people will commit updated `yarn.lock` at the same time with their dependency bumps in `package.json`, the chance of `yarn.lock` conflicts goes _down_ a lot, but admittedly not to zero. I'd totally recommend it, though.)

Yeah I'm pretty sure I made sure to commit those in my dep bumps, but will happily blame it on all the simultaneous dep bumps we have been merging today. But no worries I'll deal with them.

Otherwise as always @DeeDeeG I appreciate you reviewing the code and getting us moving. I'll go ahead and merge this one then!

**Pull request merged**